tools/check_format.py

```diff
@@ -45,25 +46,31 @@ def checkNamespace(file_path):
 def whitelistedForProtobufDeps(file_path):
   return any(path_segment in file_path for path_segment in GOOGLE_PROTOBUF_WHITELIST)
 
-def checkProtobufExternalDepsBuild(file_path):
-  if whitelistedForProtobufDeps(file_path):
-    return True
+def findSubstringAndPrintError(pattern, file_path, error_message):
   with open(file_path) as f:
-    if re.search('"protobuf"', f.read(), re.MULTILINE):
-      printError(
-          "%s has unexpected direct external dependency on protobuf, use "
-          "//source/common/protobuf instead." % file_path)
+    text = f.read()
+    if pattern in text:
+      printError(error_message)
+      for i, line in enumerate(text.splitlines()):
+        if pattern in line:
+          printError("  %s:%s" % (file_path, i + 1))
       return False
     return True
 
+def checkProtobufExternalDepsBuild(file_path):
+  if whitelistedForProtobufDeps(file_path):
+    return True
+  message = ("%s has unexpected direct external dependency on protobuf, use "
+    "//source/common/protobuf instead." % file_path)
+  return findSubstringAndPrintError('"protobuf"', file_path, message)
+
 
 def checkProtobufExternalDeps(file_path):
   if whitelistedForProtobufDeps(file_path):
     return True
   with open(file_path) as f:
     text = f.read()
-    if re.search('"google/protobuf', text, re.MULTILINE) or re.search(
-        "google::protobuf", text, re.MULTILINE):
+    if '"google/protobuf' in text or "google::protobuf" in text:
```

Comment:
is the leading quote intentional?

Reply:
I think this is intentional to detect "include" statements like https://github.com/envoyproxy/envoy/pull/1236/files#diff-c0a36114a7b25e540b0cef252d0993e3L10, since this pattern is introduced with #1236.